tests/unit/services/project-user-test.js

```diff
@@ -1,15 +1,86 @@
+import RSVP from 'rsvp';
+import { set } from '@ember/object';
 import { moduleFor, test } from 'ember-qunit';
 
 moduleFor('service:project-user', 'Unit | Service | project user', {
-  // Specify the other units that are required for this test.
   needs: [
     'service:current-user',
-    'service:flash-messages'
+    'service:flash-messages',
+    'service:session',
+    'service:metrics'
   ]
 });
 
-// Replace this with your real tests.
-test('it exists', function(assert) {
+test('it creates a new projectUser with properties', function(assert) {
+  assert.expect(3);
+  let done = assert.async();
   let service = this.subject();
-  assert.ok(service);
+  let user = { id: 1 };
+  let project = { title: 'Code Corps' };
+  let projectUser = { project, user, role: 'pending' };
+
+  let mockCurrentUser = { user };
+
+  let mockStore = {
+    createRecord(modelName, inputProperties) {
+      return {
+        save() {
+          assert.equal(modelName, 'project-user');
+          assert.deepEqual(inputProperties, projectUser);
+          return RSVP.resolve(projectUser);
+        }
+      };
+    }
+  };
+
+  let mockFlashMessage = {
+    clearMessages() {
+      return {
+        success() {
+          assert.ok(true);
+        }
+      };
+    }
+  };
+
+  set(service, 'currentUser', mockCurrentUser);
+  set(service, 'flashMessages', mockFlashMessage);
+  set(service, 'store', mockStore);
+
+  service.joinProject(project).then(() => {
+    done();
+  });
+
```

Comment:
This `test` function is still not closed by a `});` which means you're running right into the next `test` inside this `test` (which itself is not left indented correctly).